src/stability_sdk/client.py

```diff
@@ -265,38 +275,65 @@ def generate(
         else:
             seed = list(seed)
 
-        prompt_ = []
-        if isinstance(prompt, str):
-            prompt_ = [generation.Prompt(text=prompt)]
-        elif isinstance(prompt, Sequence):
-            prompt_ = [generation.Prompt(text=p) for p in prompt]
-        else:
-            raise TypeError("prompt must be a string or a sequence")
-
+        prompt_: List[generation.Prompt] = []
+        if any(isinstance(prompt, t) for t in (str, generation.Prompt)):
+            prompt = [prompt]
+        for p in prompt:
+            if isinstance(p, str):
+                p = generation.Prompt(text=p)
+            elif not isinstance(p, generation.Prompt):
+                raise TypeError("prompt must be a string or generation.Prompt object")
+            prompt_.append(p)
+
+        step_parameters = dict(
```

Comment:
this is how it was in the tester client. It might make sense to do it this way. The dict here is a way of dynamically constructing the arguments that will be used to instantiate the StepParameters object. It probably makes sense to do it this way, it ensures the defaults will get populated correctly.